lib/Doctrine/Table.php

```diff
@@ -2991,26 +3027,35 @@ public function serialize()
             $options,
             $this->_invokedMethods,
             $this->_useIdentityMap,
-        ));
+        );
     }
 
-    public function unserialize($data)
-    {
-        $all = unserialize($data);
+    /**
+     * Unserializes a Doctrine_Record instance for php 7.4+
+     *
+     * @param array $serialized
+     */
+    public function __unserialize($data) {
 
-        $this->_identifier = $all[0];
-        $this->_identifierType = $all[1];
-        $this->_columns = $all[2];
-        $this->_uniques = $all[3];
-        $this->_fieldNames = $all[4];
-        $this->_columnNames = $all[5];
-        $this->columnCount = $all[6];
-        $this->hasDefaultValues = $all[7];
-        $this->_options = $all[8];
-        $this->_invokedMethods = $all[9];
-        $this->_useIdentityMap = $all[10];
+        $this->_identifier = $data[0];
+        $this->_identifierType = $data[1];
+        $this->_columns = $data[2];
+        $this->_uniques = $data[3];
+        $this->_fieldNames = $data[4];
+        $this->_columnNames = $data[5];
+        $this->columnCount = $data[6];
+        $this->hasDefaultValues = $data[7];
+        $this->_options = $data[8];
+        $this->_invokedMethods = $data[9];
+        $this->_useIdentityMap = $data[10];
     }
 
+    /**
+     * Table loaded from cache, instead of __construct()
```

Comment:
The method name is a bit misleading.

What does it do ?

```suggestion
     * Creates new instance and initialize it from cache.
```